sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegenExec.scala

```diff
@@ -500,6 +499,19 @@ case class WholeStageCodegenRDD(@transient sc: SparkContext, var source: CodeAnd
 
   override def compute(split: Partition,
       context: TaskContext): Iterator[InternalRow] = {
+    try {
+      computeInternal(split, context)
+    } catch {
+      case ex: ClassCastException => {
+        // Any other exception in future might be added here
+        CodeGenerator.refresh(source)
+        computeInternal(split, context)
```

Comment:
I am not sure how we can identify same class cases in CCE. Also, commit or rollback will happen in partition level. Even if we retry for a false positive, it will either a complete partition or none. Isn't it?